### PR TITLE
Remove some redundancy in date filter

### PR DESCRIPTION
**Pull request by stuartskelton**
I was looking #67, I had the filtering out of the Completes, that is fairly trivial. However doing the date based filtering would mean having some duplicated code.

I started on removing some of this redundancy.

However the little hiccup and thats because  that the focus of the filtering is on the 'due' whereas I would have to have this change to be the completed.

If this is accepted I will throw up another PR for the filter by day, as it has a little complication.

As far as I can see the test suite still passes.

### CODE REVIEW

**Comment by gammons** [comment]
@stuartskelton Sorry for the delay, I'll be reviewing this today!

**Comment by stuartskelton** [comment]
No worries, I have been messing about with #75 while I was waiting.

File: todolist/date_filter.go [hunk 1]
```diff
@@ -78,6 +78,11 @@ func (f *DateFilter) filterToday(pivot time.Time) []*Todo {
 	return ret
 }
 
+
```
Comment (stuartskelton) [inline todolist/date_filter.go]: I totally agree, I just let it there for ease of review.

Comment (stuartskelton) [inline todolist/date_filter.go]: I might put it in another PR, so its easy to follow

**Pull request merged**